<?xml version="1.0" encoding="utf-8"?>

<launch>

  <!-- Load Parameter -->
  <param name="use_sim_time" value="true" />
  
  <param name="use_cpu"           value="true" />

  <!-- Start yolov5 and ros wrapper -->
  <node pkg="yolov8_ros" type="yolo_v8.py" name="yolov8_ros" output="screen" >
    <param name="weight_path"       value="$(find yolov8_ros)/weights/yolov8n.pt"/>
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/sample_weight.pt"/> -->
    <param name="image_topic"       value="/camera/color/image_raw" />
    <!-- <param name="image_topic"       value="/mirte/depth_cam/image_raw" /> -->
    <!-- <param name="depth_topic"       value="/mirte/depth_cam/depth/image_raw" /> -->
    <param name="depth_topic"       value="/camera/depth/image_raw" />
    <param name="pub_topic"         value="/yolov8/BoundingBoxes2D" />
    <param name="camera_frame"       value="camera_color_frame"/>
    <param name="visualize"       value="true"/>
    <param name="conf"              value="0.3" />
    <param name="method"              value="color_filter" />

    <param name="human_class" value="motorcycle" />
    <param name="cow_class" value="motorcycle" />
    <param name="manure_class" value="manure" />

    <param name="camera_height" value="0.1" />
  </node>

  <!-- <node name="playbag" pkg="rosbag" type="play" args="-clock /home/xaviergg/Desktop/RO_MDP/rosbag/6-new-barn-no-depth-2024-05-14-18-23-59.bag" /> -->
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find amcl)/launch/rivz.rviz"/> -->
</launch>

